### DISCUSSION TITLE
配置资源找不到index.html

### DISCUSSION THREAD

**Discussion by zhoujinshi** [AlistGo/alist#1069] [Q&A] [answered]
openwrt alist_2.5.1-1_mipsel_24kc.ipk
配置了 "assets": "https://npm.elemecdn.com/alist-web@2.5.0/dist"
错误提示 open index.html: file does not exist

**Comment by xhofe** [comment]
ipk并非由我提供，你应该去问作者。

**Reply by zhoujinshi** [reply]
用你的源码编译的
PKG_NAME:=alist
PKG_VERSION:=2.5.1
PKG_WEB_VERSION:=2.5.0
PKG_RELEASE:=$(AUTORELEASE)

PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
PKG_SOURCE_URL:=https://codeload.github.com/Xhofe/alist/tar.gz/v$(PKG_VERSION)?
PKG_HASH:=7d11b568f7a2105c25cc24ff4da89c61603a5eb2293f5d4e40fdc4b98bdfed14

**Reply by xhofe** [reply]
> 用你的源码编译的 PKG_NAME:=alist PKG_VERSION:=2.5.1 PKG_WEB_VERSION:=2.5.0 PKG_RELEASE:=$(AUTORELEASE)
> 
> PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz PKG_SOURCE_URL:=https://codeload.github.com/Xhofe/alist/tar.gz/v$(PKG_VERSION)? PKG_HASH:=7d11b568f7a2105c25cc24ff4da89c61603a5eb2293f5d4e40fdc4b98bdfed14

release下的prebuilt使用正常，源码没问题。 自己编译的问题我管不了，我也不懂ipk的编译

**Reply by zhoujinshi** [reply]
我直接使用alist-linux-musl-mipsle.tar.gz 这个是没问题的   可能使用openwrt 编译会阉割一些代码 

使用这个可以运行 但是 解压后 33M 软路由空间不够用

**Comment by wmgm183** [comment, marked as answer]
下载https://github.com/alist-org/alist-web/releases/latest/download/dist.tar.gz
放到public里面即可

**Reply by zhoujinshi** [reply]
pushd $(PKG_BUILD_DIR) ; \
			tar -zxvf $(DL_DIR)/$(PKG_NAME)-web-$(PKG_WEB_VERSION).tar.gz -C ./ ; \
			cp -ax dist/* ./public/ ; \
			cp -ax $(CURDIR)/files/assets/* ./public/assets/ ; \
			rm -rf dist/ ; \
		popd ; \



我直接解压到这个目录解决了，是github云编译出错只能下载编译好的
完整代码https://github.com/zhoujinshi/luci-packages/blob/main/alist/Makefile

**Comment by anwen-anyi** [comment]
你是不是把 **dist整个文件夹** 也放进去啦直接  哈哈~ 之前有个小伙伴编译也是这个问题 他就是直接把整个 dist父文件夹也丢进去 **public** 了,    应该是把 **dist** 里面的丢进 public~

**Comment by guanzi008** [comment]
upx压缩下
